modules/packages/Sort.chpl

```diff
@@ -591,55 +591,65 @@ proc binaryInsertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparato
  */
 proc mergeSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparator) {
   chpl_check_comparator(comparator, eltType);
-  _MergeSort(Data, minlen, comparator);
+  _MergeSort(Data, Dom.low, Dom.high+1, minlen, comparator);
 }
 
-private proc _MergeSort(Data: [?Dom], minlen=16, comparator:?rec=defaultComparator)
+private proc _MergeSort(Data: [?Dom], lo:int, hi:int, minlen=16, comparator:?rec=defaultComparator)
   where Dom.rank == 1 {
-  const lo = Dom.dim(1).low;
-  const hi = Dom.dim(1).high;
-  if hi-lo < minlen {
-    insertionSort(Data, comparator);
+  if (hi-lo < minlen) {
+    var sorted = Data[lo..(hi-1)];
+    insertionSort(sorted, comparator=comparator);
+    for (i) in (lo..(hi-1)) {
+      Data(i) = sorted(i);
```

Comment:
This copy-back is not necessary even in the slice case if you use a reference `ref sorted =  Data[lo..(hi-1)];`. But, it would be better to pass start, end into shellSort or insertionSort for the subproblem.